kroxylicious-test-tools/src/main/java/io/kroxylicious/test/tester/MockServerKroxyliciousTester.java

```diff
@@ -58,6 +70,14 @@ public Request onlyRequest() {
         return requests.get(0);
     }
 
+    public Request onlyRequestForApi(ApiKeys apiKeys) {
```

Comment:
Renamed the method.

Will merge as is and we can revisit how we set up the mocks. 

Currently if a request matches no mock response it will fail with an exception. Or do you mean register them in order? Like "i expect request A with unknownTaggedFields 1=value, fhen I expect request B`? If you do that you can make better error messages since we know what the expectation is for each call. But it might be too brittle.

We could also make a method to verify that all registered mock interactions have been invoked.